backend/api/Services/MissionDefinitionService.cs

```diff
@@ -0,0 +1,224 @@
+        "Globalization",
+        "CA1304:Specify CultureInfo",
+        Justification = "Entity framework does not support translating culture info to SQL calls"
+    )]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1307:Specify CultureInfo",
+        Justification = "Entity framework does not support translating culture info to SQL calls"
+    )]
+    public class MissionDefinitionService : IMissionDefinitionService
+    {
+        private readonly FlotillaDbContext _context;
+
+        public MissionDefinitionService(FlotillaDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<MissionDefinition> Create(MissionDefinition mission)
+        {
+            await _context.MissionDefinitions.AddAsync(mission);
+            await _context.SaveChangesAsync();
+
+            return mission;
+        }
+
+        private IQueryable<MissionDefinition> GetMissionsWithSubModels()
+        {
+            return _context.MissionDefinitions
+                .Include(mission => mission.Area)
+                .ThenInclude(robot => robot.Deck)
+                .ThenInclude(robot => robot.Installation)
+                .ThenInclude(robot => robot.Asset)
+                .Include(mission => mission.Source)
+                .Include(mission => mission.LastRun)
+                .ThenInclude(planTask => planTask == null ? null : planTask.StartTime);
+        }
+
+        public async Task<MissionDefinition?> ReadById(string id)
+        {
+            return await GetMissionsWithSubModels()
+                .FirstOrDefaultAsync(mission => mission.Id.Equals(id));
+        }
+
+        public async Task<PagedList<MissionDefinition>> ReadAll(MissionDefinitionQueryStringParameters parameters)
+        {
+            var query = GetMissionsWithSubModels();
+            var filter = ConstructFilter(parameters);
+
+            query = query.Where(filter);
+
+            SearchByName(ref query, parameters.NameSearch);
+
+            ApplySort(ref query, parameters.OrderBy);
+
+            return await PagedList<MissionDefinition>.ToPagedListAsync(
+                query,
+                parameters.PageNumber,
+                parameters.PageSize
+            );
+        }
+
+        public async Task<MissionDefinition> Update(MissionDefinition mission)
+        {
+            var entry = _context.Update(mission);
+            await _context.SaveChangesAsync();
+            return entry.Entity;
+        }
+
+        public async Task<MissionDefinition?> Delete(string id)
+        {
+            // We do not delete the source here as more than one mission definition may be using it
+            var mission = await ReadById(id);
+            if (mission is null)
+            {
+                return null;
+            }
+
+            _context.MissionDefinitions.Remove(mission);
+            await _context.SaveChangesAsync();
+
+            return mission;
+        }
```

Comment:
Should we have a delete functionality at all? Or should we instead have a field which may say that the `MissionDefinition` is deprecated or inactive? Thus we don't break anything for old `MissionRun` instances but we don't show `MissionDefinnition`objects that should not be shown?